### PR TITLE
k8s: keep KVStore CiliumNode labels synced with Node object

### PR DESCRIPTION
**Pull request by jibi**
This commit ensures that any update to the labels of a k8s Node
object is reflected in the corresponding CiliumNode KVStore entry.

### CODE REVIEW

**Comment by jibi** [comment]
/test-1.23-net-next

**Comment by jibi** [comment]
/test-1.23-net-next

**Comment by jibi** [comment]
/test-1.23-net-next

**Review by aanm — APPROVED** [review]

**Review by nathanjsweet — APPROVED** [review]

**Comment by jibi** [comment]
Reviews are in, CI is happy, just waiting on tophat approval, marking ready

**Review by tklauser — APPROVED** [review]

File: pkg/nodediscovery/nodediscovery.go
Comment (christarazi) [inline pkg/nodediscovery/nodediscovery.go:67]: Does this interface need to exist if it is already defined in `pkg/k8s/watchers/node.go`? I don't see it used in the rest of the diff for nodediscovery?

Comment (jibi) [reply pkg/nodediscovery/nodediscovery.go:67]: Right, no, this can be dropped (I have another branch which modifies this file, I'll do it there)

**Pull request merged**
